tests/ParserSpec.hs

```diff
@@ -91,13 +94,114 @@ parseNames = do
     negative "four components" $ parseQualifiedName def "a.b.c.d"
 
 parseCompile :: ParseEnv -> Text -> Either String (Either PactError [Term Name])
-parseCompile pe m = compileExps pe mkEmptyInfo <$> (parseExprs m)
+parseCompile pe m = compileExps pe mkEmptyInfo <$> parseExprs m
+
+rightRight :: Either l1 (Either l2 r) -> Bool
+rightRight (Right Right {}) = True
+rightRight _ = False
 
 narrowTryBackCompat :: Spec
 narrowTryBackCompat = do
   -- code from bad-term-in-list.repl
   it "preserves old try bug" $
-    parseCompile (ParseEnv False) "[(module m g (defcap g () 1))]" `shouldSatisfy` rightRight
+    parseCompile (ParseEnv False False) "[(module m g (defcap g () 1))]" `shouldSatisfy` rightRight
+
+condParsing :: Spec
+condParsing = do
+  it "parses single-token fallbacks" $
+    parses [text|
+      (module m g (defcap g() true)
+        (defun is-boy-or-girl:string (name:string)
+          (cond
+            ((= name "alice") "Girl")
+            ((= name "bob")   "Boy")
+            name
+          )
+        )
+      )
+      |]
+  it "parses S-expr fallbacks" $
+    parses [text|
+      (module m g (defcap g() true)
+        (defun unknown:string (name:string param:string) (format "I don't know {}" [name]))
+        (defun is-boy-or-girl:string (name:string)
+          (cond
+            ((= name "alice") "Girl")
+            ((= name "bob")   "Boy")
+            (unknown name name)
+          )
+        )
+      )
+      |]
+  it "parses S-expr fallbacks of two terms" $
+    parses [text|
+      (module m g (defcap g() true)
+        (defun unknown:string (name:string) (format "I don't know {}" [name]))
+        (defun is-boy-or-girl:string (name:string)
+          (cond
+            ((= name "alice") "Girl")
+            ((= name "bob")   "Boy")
+            (unknown name)
```

Comment:
But what happens now if `unknown` doesn't match at the end? Is this changing `cond` from an always-total form, to a now potentially partial form?

Reply:
Also, I just figured I might have got your comment wrong — the `unknown` doesn't do the matching here as it's the final term. That is, it's expected to be of type `a → b` (where `name : a`, and `b` is the type of the whole expression), while in the previous, branching terms the first one was expected to be of type `bool`.

Reply:
Ah, I see. Then this will need to be very clearly documented, since it deviates with all other Lisps, and will not be obvious at all from the rest of the construction. The way it was before, where the last element HAD to be a simple atom, was a least very clear.

Reply:
Approving when we get John's doc requests in.